pkg/config/connections.go

```diff
@@ -0,0 +1,271 @@
+package config
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/containers/storage/pkg/ioutils"
+)
+
+const connectionsFile = "podman-connections.conf"
+
+// connectionsConfigFile returns the path to the rw connections config file
+func connectionsConfigFile() (string, error) {
+	if path, found := os.LookupEnv("PODMAN_CONNECTIONS_CONF"); found {
+		return path, nil
+	}
+	path, err := userConfigPath()
+	if err != nil {
+		return "", err
+	}
+	// file is stored next to containers.conf
+	return filepath.Join(filepath.Dir(path), connectionsFile), nil
```

Comment:
What if I run `CONTAINERS_CONF=/read/only/dir/containers.conf podman ...`?  Would try to place it in the same directory?

Reply:
no it does not use CONTAINERS_CONF, it will still use the default ~/.config/containers even when CONTAINERS_CONF is set
To change the location for this new file PODMAN_CONNECTIONS_CONF must be used.